File: 03/scripts/collect.pl
#!/usr/bin/perl -w
# Process user data and update file

use CGI ":standard";

sub error {
    print "<script type=\"text/javascript\">alert('Error opening file.');window.location=\"http://162.105.146.180:8096/03/homework3-2.html\"</script>";
    exit(1);
}

sub success {
    print "<script type=\"text/javascript\">alert('Success. Thank you!');window.location=\"http://162.105.146.180:8096/03/homework3-2.html\"</script>";
}

print header();

$LOCK = 2;
$UNLOCK = 8;

my($name, $age, $gender, $email) = (param("name"), param("age"), param("gender"), param("email"));

open(DB, ">>userdata.txt") or error();
flock(DB, $LOCK);

print DB "$name\n";
print DB "$age\n";
print DB "$gender\n";
print DB "$email\n";

flock(DB, $UNLOCK);
close(DB);

success();

File: 04/collect.php
<?php
$db = mysqli_connect("localhost", "usr_2016_96", "UmQAisW2lWGwkaDFwoPB");
// account info only for test
if (!$db) {
    print "Error establishing database connection.";
    exit;
}

$err = mysqli_select_db($db, "db_2016_96");
if (!$err) {
    print "Error selecting database.";
    exit;
}

$name = mysqli_real_escape_string($db, stripslashes($_POST['name']));
$age = mysqli_real_escape_string($db, stripslashes($_POST['age']));
$gender = mysqli_real_escape_string($db, stripslashes($_POST['gender']));
$email = mysqli_real_escape_string($db, stripslashes($_POST['email']));

$query = "INSERT INTO guest(Guest_Name, Gender, Age, E_mail) VALUES('$name', '$gender', '$age', '$email');";
$result = mysqli_query($db, $query);
if (!$result) {
    print "Error querying database.";
    $error_msg = mysqli_error($db);
    print "$error_msg";
    exit;
}

?>

<script type="text/javascript">
    alert('Success. Thank you!');
    window.location="http://162.105.146.180:8096/04/homework4-2.html";
</script>
